torch/_meta_registrations.py

```diff
@@ -4316,6 +4316,82 @@ def meta_max_pool2d_with_indices(
     )
 
 
+@register_meta(aten.fractional_max_pool2d.default)
+def meta_fractional_max_pool2d(self_, kernel_size, output_size, random_samples):
+    torch._check(
+        self_.ndim in (3, 4),
+        lambda: f"fractional_max_pool2d: Expected 3D or 4D tensor, but got: {self_._ndim}",
+    )
+    ndim = self_.ndim
+
+    for i in range(3):
+        d = i + (ndim - 3)
+        torch._check(
+            self_.size(d) > 0,
+            f"fractional_max_pool2d: Expected input to have non-zero "
+            f" size for non-batch dimenions, but got {self_.size()} with dimension {d} empty",
+        )
+
+    # the check and message are out of sync, but this matches the structured meta
+    torch._check(
+        len(kernel_size) == 2,
+        lambda: "fractional_max_pool2d: kernel_size must"
+        "either be a single int or tuple of Ints",
+    )
+    torch._check(
+        len(output_size) == 2,
+        lambda: "fractional_max_pool2d: output_size must "
+        "either be a single int or tuple of Ints",
+    )
+
+    nPlane = self_.size(-3)
+    inH = self_.size(-2)
+    inW = self_.size(-1)
+    if ndim == 4:
+        nBatch = self_.size(0)
+    else:
+        nBatch = 1
+
+    samples_shape = [nBatch, nPlane, 2]
+
+    torch._check(
+        all(ss == random_samples.size(i) for i, ss in enumerate(samples_shape)),
```

Comment:
Slightly weird, but `fractional_max_pooling_check_shape` seems to allow the batch dimension to be `>=` on  `random_samples`. Also the error message for dimensional mismatch will be this `.size` call raising which isn't ideal. Eager has a an explicit dim check.